diff --git a/tools/firmware/Makefile b/tools/firmware/Makefile
index 5458017..2e37755 100644
--- a/tools/firmware/Makefile
+++ b/tools/firmware/Makefile
@@ -25,6 +25,9 @@ ovmf-dir:
 	sed 's|python |python2 |g' -i "ovmf-dir/BaseTools/Tests/GNUmakefile"
 	patch -Np1 -i ovmf-gcc4.9-basetools.patch
 	patch -Np1 -i ovmf-gcc4.9-ovmfpkg.patch
+	patch -Np1 -i ovmf-gcc5-basetools.patch
+	patch -Np1 -i ovmf-gcc5-ovmfpkg.patch
+	patch -Np1 -i ovmf-vfrcompiler.patch
 	cp ovmf-makefile ovmf-dir/Makefile;
 
 seabios-dir:
