diff --git a/BaseTools/Conf/tools_def.template b/BaseTools/Conf/tools_def.template
index 739c33d..49e669d 100644
--- a/ovmf-dir/BaseTools/Conf/tools_def.template
+++ b/ovmf-dir/BaseTools/Conf/tools_def.template
@@ -165,6 +165,9 @@ DEFINE GCC48_X64_PREFIX        = /usr/bin/
 DEFINE GCC49_IA32_PREFIX       = /usr/bin/
 DEFINE GCC49_X64_PREFIX        = /usr/bin/
 
+DEFINE GCC5_IA32_PREFIX       = /usr/bin/
+DEFINE GCC5_X64_PREFIX        = /usr/bin/
+
 DEFINE UNIX_IASL_BIN           = /usr/bin/iasl
 #DEFINE UNIX_IASL_BIN           = $(HOME)/programs/iasl
 DEFINE WIN_ASL_BIN_DIR         = C:\ASL
@@ -316,6 +319,12 @@ DEFINE CLANG_BIN = /usr/bin/
 #                             Required to build platforms or ACPI tables:
 #                               Intel(r) ACPI Compiler v20101013 from
 #                               http://www.acpica.org/downloads/previous_releases.php
+#   GCC5       -Linux-  Requires:
+#                             GCC 5.x
+#                        Optional:
+#                             Required to build platforms or ACPI tables:
+#                               Intel(r) ACPI Compiler v20101013 from
+#                               http://www.acpica.org/downloads/previous_releases.php
 #   ELFGCC      -Linux-  Requires:
 #                             GCC(this tool chain uses whatever version of gcc and binutils that is installed in /usr/bin)
 #                        Optional:
@@ -3220,6 +3229,22 @@ DEFINE GCC49_AARCH64_DLINK_FLAGS     = DEF(GCC_ARM_AARCH64_DLINK_COMMON)
 DEFINE GCC49_ARM_ASLDLINK_FLAGS      = DEF(GCC_ARM_AARCH64_ASLDLINK_FLAGS) --oformat=elf32-littlearm
 DEFINE GCC49_AARCH64_ASLDLINK_FLAGS  = DEF(GCC_ARM_AARCH64_ASLDLINK_FLAGS)
 
+DEFINE GCC5_IA32_CC_FLAGS           = DEF(GCC49_IA32_CC_FLAGS) -std=gnu89
+DEFINE GCC5_X64_CC_FLAGS            = DEF(GCC49_X64_CC_FLAGS) -std=gnu89
+DEFINE GCC5_IA32_X64_DLINK_COMMON   = -nostdlib -n -q --gc-sections --script=$(EDK_TOOLS_PATH)/Scripts/gcc4.9-ld-script
+DEFINE GCC5_IA32_X64_ASLDLINK_FLAGS = DEF(GCC5_IA32_X64_DLINK_COMMON) --entry ReferenceAcpiTable -u ReferenceAcpiTable
+DEFINE GCC5_IA32_X64_DLINK_FLAGS    = DEF(GCC5_IA32_X64_DLINK_COMMON) --entry $(IMAGE_ENTRY_POINT) -u $(IMAGE_ENTRY_POINT) -Map $(DEST_DIR_DEBUG)/$(BASE_NAME).map
+DEFINE GCC5_X64_DLINK_FLAGS         = DEF(GCC5_IA32_X64_DLINK_FLAGS)  -melf_x86_64 --oformat=elf64-x86-64
+DEFINE GCC5_ASM_FLAGS               = DEF(GCC49_ASM_FLAGS)
+DEFINE GCC5_ARM_ASM_FLAGS           = DEF(GCC49_ARM_ASM_FLAGS)
+DEFINE GCC5_AARCH64_ASM_FLAGS       = DEF(GCC49_AARCH64_ASM_FLAGS)
+DEFINE GCC5_ARM_CC_FLAGS            = DEF(GCC49_ARM_CC_FLAGS)
+DEFINE GCC5_AARCH64_CC_FLAGS        = DEF(GCC49_AARCH64_CC_FLAGS)
+DEFINE GCC5_ARM_DLINK_FLAGS         = DEF(GCC49_ARM_DLINK_FLAGS)
+DEFINE GCC5_AARCH64_DLINK_FLAGS     = DEF(GCC_ARM_AARCH64_DLINK_COMMON)
+DEFINE GCC5_ARM_ASLDLINK_FLAGS      = DEF(GCC49_ARM_ASLDLINK_FLAGS)
+DEFINE GCC5_AARCH64_ASLDLINK_FLAGS  = DEF(GCC_ARM_AARCH64_ASLDLINK_FLAGS)
+
 ####################################################################################
 #
 # Unix GCC And Intel Linux ACPI Compiler
@@ -3767,6 +3792,129 @@ RELEASE_GCC49_AARCH64_CC_FLAGS   = DEF(GCC49_AARCH64_CC_FLAGS) -Wno-unused-but-s
 
 ####################################################################################
 #
+# GCC 5.x - This configuration is used to compile under Linux to produce
+#           PE/COFF binaries using GCC 5.x.
+#
+####################################################################################
+*_GCC5_*_*_FAMILY               = GCC
+
+*_GCC5_*_MAKE_PATH                    = DEF(GCC5_IA32_PREFIX)make
+*_GCC5_*_*_DLL                        = ENV(GCC5_DLL)
+*_GCC5_*_ASL_PATH                     = DEF(UNIX_IASL_BIN)
+
+*_GCC5_*_PP_FLAGS                     = DEF(GCC_PP_FLAGS)
+*_GCC5_*_ASLPP_FLAGS                  = DEF(GCC_ASLPP_FLAGS)
+*_GCC5_*_ASLCC_FLAGS                  = DEF(GCC_ASLCC_FLAGS)
+*_GCC5_*_VFRPP_FLAGS                  = DEF(GCC_VFRPP_FLAGS)
+*_GCC5_*_APP_FLAGS                    =
+*_GCC5_*_ASL_FLAGS                    = DEF(IASL_FLAGS)
+*_GCC5_*_ASL_OUTFLAGS                 = DEF(IASL_OUTFLAGS)
+
+##################
+# GCC5 IA32 definitions
+##################
+*_GCC5_IA32_OBJCOPY_PATH         = DEF(GCC5_IA32_PREFIX)objcopy
+*_GCC5_IA32_CC_PATH              = DEF(GCC5_IA32_PREFIX)gcc
+*_GCC5_IA32_SLINK_PATH           = DEF(GCC5_IA32_PREFIX)ar
+*_GCC5_IA32_DLINK_PATH           = DEF(GCC5_IA32_PREFIX)ld
+*_GCC5_IA32_ASLDLINK_PATH        = DEF(GCC5_IA32_PREFIX)ld
+*_GCC5_IA32_ASM_PATH             = DEF(GCC5_IA32_PREFIX)gcc
+*_GCC5_IA32_PP_PATH              = DEF(GCC5_IA32_PREFIX)gcc
+*_GCC5_IA32_VFRPP_PATH           = DEF(GCC5_IA32_PREFIX)gcc
+*_GCC5_IA32_ASLCC_PATH           = DEF(GCC5_IA32_PREFIX)gcc
+*_GCC5_IA32_ASLPP_PATH           = DEF(GCC5_IA32_PREFIX)gcc
+*_GCC5_IA32_RC_PATH              = DEF(GCC5_IA32_PREFIX)objcopy
+
+*_GCC5_IA32_ASLCC_FLAGS          = DEF(GCC_ASLCC_FLAGS) -m32
+*_GCC5_IA32_ASLDLINK_FLAGS       = DEF(GCC5_IA32_X64_ASLDLINK_FLAGS) -m elf_i386
+*_GCC5_IA32_ASM_FLAGS            = DEF(GCC5_ASM_FLAGS) -m32 -march=i386
+*_GCC5_IA32_CC_FLAGS             = DEF(GCC5_IA32_CC_FLAGS) -Os
+*_GCC5_IA32_DLINK_FLAGS          = DEF(GCC5_IA32_X64_DLINK_FLAGS) -m elf_i386 --oformat=elf32-i386
+*_GCC5_IA32_RC_FLAGS             = DEF(GCC_IA32_RC_FLAGS)
+*_GCC5_IA32_OBJCOPY_FLAGS        = 
+*_GCC5_IA32_NASM_FLAGS           = -f elf32
+
+##################
+# GCC5 X64 definitions
+##################
+*_GCC5_X64_OBJCOPY_PATH         = DEF(GCC5_X64_PREFIX)objcopy
+*_GCC5_X64_CC_PATH              = DEF(GCC5_X64_PREFIX)gcc
+*_GCC5_X64_SLINK_PATH           = DEF(GCC5_X64_PREFIX)ar
+*_GCC5_X64_DLINK_PATH           = DEF(GCC5_X64_PREFIX)ld
+*_GCC5_X64_ASLDLINK_PATH        = DEF(GCC5_X64_PREFIX)ld
+*_GCC5_X64_ASM_PATH             = DEF(GCC5_X64_PREFIX)gcc
+*_GCC5_X64_PP_PATH              = DEF(GCC5_X64_PREFIX)gcc
+*_GCC5_X64_VFRPP_PATH           = DEF(GCC5_X64_PREFIX)gcc
+*_GCC5_X64_ASLCC_PATH           = DEF(GCC5_X64_PREFIX)gcc
+*_GCC5_X64_ASLPP_PATH           = DEF(GCC5_X64_PREFIX)gcc
+*_GCC5_X64_RC_PATH              = DEF(GCC5_X64_PREFIX)objcopy
+
+*_GCC5_X64_ASLCC_FLAGS          = DEF(GCC_ASLCC_FLAGS) -m64
+*_GCC5_X64_ASLDLINK_FLAGS       = DEF(GCC5_IA32_X64_ASLDLINK_FLAGS) -m elf_x86_64
+*_GCC5_X64_ASM_FLAGS            = DEF(GCC5_ASM_FLAGS) -m64
+*_GCC5_X64_CC_FLAGS             = DEF(GCC5_X64_CC_FLAGS)
+*_GCC5_X64_DLINK_FLAGS          = DEF(GCC5_X64_DLINK_FLAGS)
+*_GCC5_X64_RC_FLAGS             = DEF(GCC_X64_RC_FLAGS)
+*_GCC5_X64_OBJCOPY_FLAGS        = 
+*_GCC5_X64_NASM_FLAGS           = -f elf64
+
+##################
+# GCC5 ARM definitions
+##################
+*_GCC5_ARM_CC_PATH              = ENV(GCC5_ARM_PREFIX)gcc
+*_GCC5_ARM_SLINK_PATH           = ENV(GCC5_ARM_PREFIX)ar
+*_GCC5_ARM_DLINK_PATH           = ENV(GCC5_ARM_PREFIX)ld
+*_GCC5_ARM_ASLDLINK_PATH        = ENV(GCC5_ARM_PREFIX)ld
+*_GCC5_ARM_ASM_PATH             = ENV(GCC5_ARM_PREFIX)gcc
+*_GCC5_ARM_PP_PATH              = ENV(GCC5_ARM_PREFIX)gcc
+*_GCC5_ARM_VFRPP_PATH           = ENV(GCC5_ARM_PREFIX)gcc
+*_GCC5_ARM_ASLCC_PATH           = ENV(GCC5_ARM_PREFIX)gcc
+*_GCC5_ARM_ASLPP_PATH           = ENV(GCC5_ARM_PREFIX)gcc
+*_GCC5_ARM_RC_PATH              = ENV(GCC5_ARM_PREFIX)objcopy
+
+*_GCC5_ARM_ARCHCC_FLAGS         = -mthumb
+*_GCC5_ARM_PLATFORM_FLAGS       = -march=armv7-a
+
+*_GCC5_ARM_ASLCC_FLAGS          = DEF(GCC_ASLCC_FLAGS)
+*_GCC5_ARM_ASLDLINK_FLAGS       = DEF(GCC5_ARM_ASLDLINK_FLAGS)
+*_GCC5_ARM_ASM_FLAGS            = DEF(GCC5_ARM_ASM_FLAGS)
+*_GCC5_ARM_DLINK_FLAGS          = DEF(GCC5_ARM_DLINK_FLAGS)
+*_GCC5_ARM_PLATFORM_FLAGS       = -march=armv7-a
+*_GCC5_ARM_PP_FLAGS             = $(ARCHCC_FLAGS) $(PLATFORM_FLAGS) DEF(GCC_PP_FLAGS)
+*_GCC5_ARM_RC_FLAGS             = DEF(GCC_ARM_RC_FLAGS)
+*_GCC5_ARM_VFRPP_FLAGS          = $(ARCHCC_FLAGS) $(PLATFORM_FLAGS) DEF(GCC_VFRPP_FLAGS)
+
+  DEBUG_GCC5_ARM_CC_FLAGS       = DEF(GCC5_ARM_CC_FLAGS) -O0
+RELEASE_GCC5_ARM_CC_FLAGS       = DEF(GCC5_ARM_CC_FLAGS) -Wno-unused-but-set-variable
+
+##################
+# GCC5 AARCH64 definitions
+##################
+*_GCC5_AARCH64_CC_PATH          = ENV(GCC5_AARCH64_PREFIX)gcc
+*_GCC5_AARCH64_SLINK_PATH       = ENV(GCC5_AARCH64_PREFIX)ar
+*_GCC5_AARCH64_DLINK_PATH       = ENV(GCC5_AARCH64_PREFIX)ld
+*_GCC5_AARCH64_ASLDLINK_PATH    = ENV(GCC5_AARCH64_PREFIX)ld
+*_GCC5_AARCH64_ASM_PATH         = ENV(GCC5_AARCH64_PREFIX)gcc
+*_GCC5_AARCH64_PP_PATH          = ENV(GCC5_AARCH64_PREFIX)gcc
+*_GCC5_AARCH64_VFRPP_PATH       = ENV(GCC5_AARCH64_PREFIX)gcc
+*_GCC5_AARCH64_ASLCC_PATH       = ENV(GCC5_AARCH64_PREFIX)gcc
+*_GCC5_AARCH64_ASLPP_PATH       = ENV(GCC5_AARCH64_PREFIX)gcc
+*_GCC5_AARCH64_RC_PATH          = ENV(GCC5_AARCH64_PREFIX)objcopy
+
+*_GCC5_AARCH64_ASLCC_FLAGS      = DEF(GCC_ASLCC_FLAGS)
+*_GCC5_AARCH64_ASLDLINK_FLAGS   = DEF(GCC5_AARCH64_ASLDLINK_FLAGS)
+*_GCC5_AARCH64_ASM_FLAGS        = DEF(GCC5_AARCH64_ASM_FLAGS)
+*_GCC5_AARCH64_DLINK_FLAGS      = DEF(GCC5_AARCH64_DLINK_FLAGS)
+*_GCC5_AARCH64_PLATFORM_FLAGS   =
+*_GCC5_AARCH64_PP_FLAGS         = $(ARCHCC_FLAGS) $(PLATFORM_FLAGS) DEF(GCC_PP_FLAGS)
+*_GCC5_AARCH64_RC_FLAGS         = DEF(GCC_AARCH64_RC_FLAGS)
+*_GCC5_AARCH64_VFRPP_FLAGS      = $(ARCHCC_FLAGS) $(PLATFORM_FLAGS) DEF(GCC_VFRPP_FLAGS)
+
+  DEBUG_GCC5_AARCH64_CC_FLAGS   = DEF(GCC5_AARCH64_CC_FLAGS) -O0
+RELEASE_GCC5_AARCH64_CC_FLAGS   = DEF(GCC5_AARCH64_CC_FLAGS) -Wno-unused-but-set-variable
+
+####################################################################################
+#
 # Cygwin GCC And Intel ACPI Compiler
 #
 ####################################################################################
